lib/matplotlib/tests/test_colorbar.py

```diff
@@ -308,3 +310,34 @@ def test_colorbar_lognorm_extension():
     cb = ColorbarBase(ax, norm=LogNorm(vmin=0.1, vmax=1000.0),
                       orientation='vertical', extend='both')
     assert cb._values[0] >= 0.0
+
+
+@image_comparison(baseline_images=['extended_cbar_with_contourf_min',
+                                   'extended_cbar_with_contourf_max',
+                                   'extended_cbar_with_contourf_both'],
+                  extensions=['png'])
+def test_extended_colorbar_on_contourf():
+    np.random.seed(1)
+    N = 100
+    x = np.linspace(-3.0, 3.0, N)
+    y = np.linspace(-2.0, 2.0, N)
+    X, Y = np.meshgrid(x, y)
+    z = (bivariate_normal(X, Y, 0.1, 0.2, 1.0, 1.0)
```

Comment:
For the purpose of a test you can use a simpler pattern with far fewer points.  For example, you could use
```python
z = np.logspace(0.1, 10, 24).reshape((4, 6))
```
and omit the X and Y entirely.